src/mpid/ch4/shm/ipc/src/ipc_win.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpidimpl.h"
+#include "ipc_noinline.h"
+#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
+#include "../xpmem/xpmem_noinline.h"
+#endif
+
+int MPIDI_IPC_mpi_win_create_hook(MPIR_Win * win)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_IPC_MPI_WIN_CREATE_HOOK);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IPC_MPI_WIN_CREATE_HOOK);
+
+#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
+    if (MPIR_CVAR_CH4_XPMEM_LMT_MSG_SIZE != -1) {
+        mpi_errno = MPIDI_IPC_xpmem_mpi_win_create_hook(win);
```

Comment:
It's unclear why this function exists.  Why does XPMEM need to know that this is a window?  Doesn't it simply need to open a memory handle to some buffer and simply return it?  That fact that the buffer belongs to a window seems irrelevant here.

Reply:
We didn't think too much about the RMA routine. Maybe you are right. Let me try it.